planet/clients/orders.py

```diff
@@ -313,6 +338,26 @@ async def download_order(self,
                                       progress_bar=progress_bar)
             for location in locations
         ]
+        if checksum:
+            # Checksum Implementation
+            # Get manifest filepath
+            manifest_json = ' '.join([x for x in filenames if x.endswith('manifest.json')])
+            # Save each filename and respective hash in a dict as a key-value pair
+            with open(manifest_json, 'rb') as manifest:
+                manifest_data = json.load(manifest)
+                file_key_pairs = {}
+                for json_entry in manifest_data['files']:
+                    file_name = json_entry['path'].split('/')[-1]
+                    origin_hash = json_entry['digests'][checksum.lower()]
+                    file_key_pairs[file_name] = origin_hash
+            # For each file (not including manifest json file), calculate hash on its contents
+            filenames_loop = [x for x in filenames if not x.endswith('manifest.json')]
```

Comment:
For your test, you currently only provide a `manifest.json`, so this will return an empty string and you'll never run `self.calculate_checksum()`.